src/comment_worker.py

```diff
@@ -265,20 +265,19 @@ def invest(self, sess, comment, investor, amount, suffix):
         """
         This function invests
         """
-        if config.PREVENT_INSIDERS:
-            if comment.submission.author.name == comment.author.name:
-                return comment.reply_wrap(message.INSIDE_TRADING_ORG)
-        multiplier = CommentWorker.multipliers.get(suffix, 1)
+        if config.PREVENT_INSIDERS and comment.submission.author.name == comment.author.name:
+            return comment.reply_wrap(message.INSIDE_TRADING_ORG)
 
         # Allows input such as '!invest 100%' and '!invest 50%'
-        if multiplier == '%':
-            amount = int(investor.balance * (int(amount)/100))
-        else:
-            try:
-                amount = int(amount.replace(',', ''))
-                amount = amount * multiplier
-            except ValueError:
-                return
+        multiplier = CommentWorker.multipliers.get(suffix, 1)
+        try:
+            input_amount = int(amount.replace(',', ''))
+            if multiplier == '%':
+                multiplier = float(input_amount/100) if amount < 100 else 1
+                input_amount = int(investor.balance)
+            amount = input_amount * multiplier
```

Comment:
~Do you think we would need to set the multiplier value here to 1 if it doesn't equal to `%`?~

Missed a part. Good stuff